File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    

    Min = zero = 0
    rank_lst = {6:1 , 5:2 , 4:3 , 3:4 , 2:5}
    for i in lottos:
        if i in win_nums:Min += 1
        if i == 0 : zero += 1
    Max = Min + zero
    answer = [rank_lst.get(Max,6),rank_lst.get(Min,6)]
    
    
    return answer

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    const rank = { 6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}
    let correct = 0, zero = 0
    
    lottos.forEach((val) => {
        if(val === 0) { zero += 1}
        else if (win_nums.includes(val)) { correct += 1 }
    })
    
    return [rank[correct + zero],rank[correct]];
}